akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -44,23 +44,31 @@ abstract class HttpApp extends Directives {
   }
 
   /**
-   * Start a server on the specified host and port, using provided settings.
+   * Start a server on the specified host and port, using the provided [[ActorSystem]].
+   * Note that this method is blocking
+   */
+  def startServer(host: String, port: Int, system: ActorSystem): Unit = {
+    startServer(host, port, ServerSettings(system), Some(system))
```

Comment:
to be consistent with JavaDSL and also do what we want here, this should be `Option(system)` or we should explicitly validate it's not null